packages/next/src/lib/metadata/resolve-title.ts

```diff
@@ -0,0 +1,27 @@
+import type { Metadata } from './types/metadata-interface'
+import type { AbsoluteTemplateString } from './types/metadata-types'
+
+function resolveTitleTemplate(template: string | null, title: string) {
+  return template ? template.replace(/%s/g, title) : title
+}
+
+export function resolveTitle(
+  stashed: AbsoluteTemplateString,
+  title: Metadata['title']
+) {
+  const resolved: AbsoluteTemplateString = { ...stashed }
+  if (typeof title === 'string') {
+    resolved.absolute = resolveTitleTemplate(stashed.template, title)
+  } else if (title) {
+    if ('default' in title) {
+      resolved.absolute = resolveTitleTemplate(stashed.template, title.default)
+    }
+    if ('absolute' in title) {
+      resolved.absolute = title.absolute
+    }
+    if (title && 'template' in title && title.template) {
+      resolved.template = title.template
+    }
```

Comment:
```suggestion
    if ('template' in title) {
      resolved.template = title.template
    }
```
we don't want to truthy check the template b/c otherwise you could never "unset" it for this deeper layers. I guess you could pass `"%s"` which is kind of like "no template".

I wonder if our types need to change a little bit for titles.

Having `AbsoluteTemplateString` always be set for `ResolvedMetadata['title']` is nice because you don't have to do much type checking to refine the type information, it's always just known.

However it makes uses of an awkward type cast for "no template"

To convey no configured title template we have `{ absolute: "...", template: "" }`

But if one wants to set the template to "no template" it seems odd to me that they have to pass an empty string. It would be nice if they could pass `{ absolute: "...", template: null }`